merkle/hashers/tree_hasher.go

```diff
@@ -42,3 +48,47 @@ type MapHasher interface {
 	// TODO(gbelvin): Replace Size() with BitLength().
 	Size() int
 }
+
+var (
+	maxHashers = trillian.HashStrategy(len(trillian.HashStrategy_value))
+	logHashers = make([]LogHasher, maxHashers)
+	mapHashers = make([]MapHasher, maxHashers)
+)
+
+// RegisterLogHasher registers a hasher for use.
+func RegisterLogHasher(h trillian.HashStrategy, f LogHasher) {
+	if h < 0 || h >= maxHashers {
+		panic(fmt.Sprintf("RegisterLogHasher(%v) of unknown hasher", h))
```

Comment:
Good point. should I remove the check? 

I'm following the style of golang's crypto package here. Since this function should only be called in an init method, I assumed that a panic was more ok (ish). Ie. it should only be triggered right on launch.